wolfhsm/wh_server.h

```diff
@@ -2,18 +2,51 @@
 #define WOLFHSM_WH_SERVER_H_
 
 #include "wolfhsm/wh_comm.h"
+#include "wolfhsm/wh_common.h"
+#include "wolfhsm/common.h"
+
+#ifndef WOLFSSL_USER_SETTINGS
+    #include "wolfssl/options.h"
+#endif
+#include "wolfssl/wolfcrypt/settings.h"
+#include "wolfssl/wolfcrypt/random.h"
+#include "wolfssl/wolfcrypt/curve25519.h"
 
 #if 0
 #include "wolfhsm/nvm.h"
 #include "wolfhsm/nvm_remote.h"
 #endif
 
+typedef struct NvmMetaData {
+    uint16_t id;
+    uint32_t flags;
+    uint16_t len;
+    uint8_t label[WOLFHSM_NVM_LABEL_LEN];
+    uint8_t confDigest[WOLFHSM_DIGEST_STUB];
+#ifdef WOLFHSM_SHE_EXTENSION
+    uint32_t count;
+#endif
+} NvmMetaData;
+
+typedef struct CacheSlot {
+    uint8_t commited;
+    NvmMetaData meta[1];
+    uint8_t buffer[WOLFHSM_NVM_MAX_OBJECT_SIZE];
```

Comment:
This is not going to work on small RAM devices.  Max object size is 64kB.  The intent would be to dynamically load whatever content is needed during the request and unload it afterwards.  The NVM Read function allows partial reads at an offset so only the pertinent part of the object data must be loaded in RAM at any given time.  Let's work on an alternate cache design that doesn't necessarily rely on fully preallocating RAM storage.  Ok for now, but this is a design issue.

Reply:
Added to todo list